### PR TITLE
Issue 94 create profile http api

### LINKED ISSUE #94: Create Profile HTTP API
## Basic Information
Design a API interface for profile table, so profile data in database can be accessed by API call.

### PR DESCRIPTION
**Pull request by TonyyZzz**
## Change Summary
**Create API for Profile table, so front end can retrieve, update and create profile instance to database.**

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [x] The change has documentation

## Other Information
**Added a new library `Pillow` for image upload, Please rebuild docker container by running `docker composer build` and restart docker container**

# Related issue

- Resolve #94

### CODE REVIEW

File: server/app/migrations/0003_alter_profiles_avatar_url.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 5.0.6 on 2024-07-17 09:49
+
+import app.models
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('app', '0002_alter_profiles_avatar_url'),
```
Comment (yunho7687) [inline server/app/migrations/0003_alter_profiles_avatar_url.py:10]: It seems that you forget to add '0002_alter_profiles_avatar_url' to commit

Comment (TonyyZzz) [reply server/app/migrations/0003_alter_profiles_avatar_url.py:10]: image
0002_alter_profile_avatar_url is here

File: server/poetry.lock [hunk 2]
```diff
@@ -268,6 +268,103 @@ files = [
     {file = "packaging-24.1.tar.gz", hash = "sha256:026ed72c8ed3fcce5bf8950572258698927fd1dbda10a5e981cdf0ac37f4f002"},
 ]
 
+[[package]]
+name = "pillow"
+version = "10.4.0"
+description = "Python Imaging Library (Fork)"
+optional = false
+python-versions = ">=3.8"
+files = [
```
Comment (yunho7687) [inline server/poetry.lock:291]: Could you explain what this is and what it does

Comment (TonyyZzz) [reply server/poetry.lock:291]: image
In models.py, I used ImageField for avatar_url, this is where 'Pillow' package required.

**Review by 0XFF-96 — APPROVED** [review]

**Comment by 0XFF-96** [comment]
@yunho7687

**Pull request merged**